mantis-runtime/src/main/java/io/mantisrx/runtime/ScalarToGroup.java

```diff
@@ -116,6 +118,12 @@ public Config<T, K, R> concurrentInput() {
             return this;
         }
 
+        public Config<T, K, R> concurrentInput(final int concurrency) {
```

Comment:
this could cause issues, no?

Reply:
I went looking around and we do [allocate the same key to the same thread every time](https://github.com/Netflix/mantis/blob/a59f5f659ad105e0c5aea948dadd748655c09911/mantis-runtime/src/main/java/io/mantisrx/runtime/executor/StageExecutors.java#L238). We have concurrency either way, this just lets the user specify it precisely, unless I'm misunderstanding the if/else that contains the linked line.